core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1158,9 +1158,14 @@ class KafkaController(val config : KafkaConfig, zkUtils: ZkUtils, val brokerStat
     @throws(classOf[Exception])
     def handleNewSession() {
       info("ZK expired; shut down all controller components and try to re-elect")
-      onControllerResignation()
-      inLock(controllerContext.controllerLock) {
-        controllerElector.elect
+      if (controllerElector.getControllerID() != config.brokerId) {
+        onControllerResignation()
+        inLock(controllerContext.controllerLock) {
+          controllerElector.elect
+        }
+      } else {
```

Comment:
@junrao  It's the case 1.  ephemeral node is created by LeaderChangeListener.handleDataDeleted, but in KafkaController.handleNewSession will recreate the node but throw ZKNodeExistException